docs/DeveloperGuide.md

```diff
@@ -1,22 +1,118 @@
+
+The `TicTacToe` class:
+* reads commands using `Scanner`, after which the input is parsed by the `Parser` class. 
+  * Specifically, it takes in values only from `1` to `9`.
+* stores the `board` upon which the game is played on. 
+  * The `board` is a `String` `array` with `9` slots, indicating the possible inputs from the user.
+  * The `board` is also printed at the start of the game and after every move by the user.
+* checks the winner using the `checkWinner(int turnCount)` method, which takes in the number of turns taken so far.
+  * Returns the following based on the state of the game:
+    * `unending` if the game is unfinished or the user inputs `quit`.
+    * `X` if the user wins the game
+    * `O` if the user loses the game
+    * `draw` if the user and the computer draw
+
+
+### <span style="color:grey;">Hangman Implementation:</span>
+The `HangMan` class:
+* When the game is started, `runHangMan` is called.
+* User input is read using `Scanner`. 
+  * Available commands currently are only limited to single characters.
+    * (A future implementation for v2.0 would be to additionally include the ability to input an entire word for guessing)
+* Stores a `wordBank` which contains the list of available words randomly selected for the hangman game.
+* Contains attributes of `chosenWordLength` based on the randomly `chosenWord`.
+* User inputs are compared against the `chosenWord` to check for similar letters used.
+   Each iteration of user input will result in calls of `addGuess`, `printHangMan`, `printLettersGuessed` & `printWordGuesser`.
+  * `addGuess` adds the user input to an arraylist<String> of `allGuessedLetters`
+    * `printLettersGuessed` then iterates through `allGuessedLetters`, printing each element of the array list.
+  * Correctly guessed letters together with `_` blank spaces corresponding to letters not yet guessed, are printed using `printWordGuesser`.
+  * Incorrect guesses however lead to the game `state` increasing by 1.
+    * `printHangMan` prints a different hangman based on the game `state`. 
+    * Should the `state` reach 6, the game is lost.
+* `runHangMan` checks for the number of `_` blank spaces left at each iteration of user input.
+  * Should the number of `_` be 0, the player has won.
 
 ## Product scope
-### Target user profile
 
-{Describe the target user profile}
+### <span style="color:grey;">Target user profile</span>
+
+- Loves to play games and challenge themselves
+- Prefers using desktop for entertainment
+- Has particular interest in CLI commands
+- Enjoys retro-style
 
-### Value proposition
+### <span style="color:grey;">Value proposition</span>
 
-{Describe the value proposition: what problem does it solve?}
+CLI-based games work on different operating systems and environments that support 
+command-line interfaces. This means players can enjoy the game no matter what kind 
+of computer or setup they use.
```

Comment:
I think u are trying to say that ur app is a CLI based game that works on different operating system, so could perhaps paraphrase to "CLI-based games that can ..."